types/shares.go

```diff
@@ -72,17 +73,19 @@ func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte)
 
 // splitContiguous splits multiple raw data contiguously as shares.
 // Used for transactions, intermediate state roots, and evidence.
-func splitContiguous(nid namespace.ID, rawDatas [][]byte) []NamespacedShare {
+func splitContiguous(id namespace.ID, rawDatas [][]byte) []NamespacedShare {
 	shares := make([]NamespacedShare, 0)
 	// Index into the outer slice of rawDatas
 	outerIndex := 0
 	// Index into the inner slice of rawDatas
 	innerIndex := 0
 	for outerIndex < len(rawDatas) {
 		var rawData []byte
+		nid := make([]byte, len(id))
+		copy(nid, id)
 		startIndex := 0
 		rawData, outerIndex, innerIndex, startIndex = getNextChunk(rawDatas, outerIndex, innerIndex, TxShareSize)
-		rawShare := []byte(append(append(nid, byte(startIndex)), rawData...))
+		rawShare := append(append(nid, byte(startIndex)), rawData...)
```

Comment:
Same as above.  The proper use of append could increase the readability. 

Also in this case here, we know exactly how large the resulting slide will be (namely `len(nid)+1+len(rawData)`). Using the snippet below should also more efficient in terms of allocations.

```go
rawShare := append(append(append(
	make([]byte, len(nid)+1+len(rawData)), 
	nid...), 
	byte(startIndex)), 
	rawData...)
```

Reply:
sorry, I think I lost a draft of a comment somewhere and ended up not explaining why I added the utility function. I don't think using append in this way will serve our purposes.
https://play.golang.org/p/NN4V2gaLh8k